### PR TITLE
MTP-698_too_many_characters_feedback

### PR DESCRIPTION
**Pull request by amundlindstol**

### CODE REVIEW

File: src/svelte/components/form/TextInput.svelte [hunk 1]
```diff
@@ -60,6 +64,9 @@
       id={name}
       {name}
       use:countCharacters={countCharsParams}
+      on:input={e => {
```
Comment (lha047) [inline src/svelte/components/form/TextInput.svelte:67]: Screenshot 2023-02-28 at 16 07 01

Something doesnt look right here :P Ask design how it should be

**Review by lha047 — CHANGES_REQUESTED** [review]

File: src/svelte/components/form/TextInput.svelte [hunk 2]
```diff
@@ -1,6 +1,7 @@
+<!--suppress XmlDuplicatedId -->
```
Comment (lha047) [inline src/svelte/components/form/TextInput.svelte:1]: What is this? 🤔

File: src/svelte/components/form/TextArea.svelte [hunk 3]
```diff
@@ -49,10 +52,14 @@
   <InputError {...error} {hiddenErrorText} />
 {/if}
 
+<!--suppress CheckEmptyScriptTag -->
```
Comment (lha047) [inline src/svelte/components/form/TextArea.svelte:55]: What is this? 🤔

Comment (amundlindstol) [reply src/svelte/components/form/TextArea.svelte:55]: Intellij inspection complained on empty tag </> vs open tag <></>
But apparently Netscape 4 is the only browser not supporting it.
so i disabled HTML | Empty tag inspection in Settings | Editor | Inspections

File: src/ts/count-characters.ts [hunk 4]
```diff
@@ -1,4 +1,20 @@
-import type {Action} from './types'
+import type {Action, ErrorDetail} from './types'
+
+export function errorOnTooManyCharacters(
+  e: InputEvent,
+  countCharsParams: Record<string, any>,
```
Comment (lha047) [inline src/ts/count-characters.ts:5]: Maybe create at proper type for this. Dont like the any part of the record :P

File: src/scss/global/_typography.scss [hunk 5]
```diff
@@ -161,6 +161,9 @@ $content-selectors: '.content' !default;
 strong {
   font-weight: var(--font-weight-bold);
 }
+.text-error {
```
Comment (lha047) [inline src/scss/global/_typography.scss:164]: Dont need this. There is a `color-error` class that does the exact same thing

File: src/ts/count-characters.ts [hunk 6]
```diff
@@ -36,8 +52,10 @@ export const countCharacters: Action<HTMLInputElement | HTMLTextAreaElement> = (
     const countCharactersLeft = params.maxlength - node.value.length
     if (countCharactersLeft < 0) {
       counterElVisible.innerText = `${-countCharactersLeft} ${charactersTooMany}`
+      counterElVisible.classList.add('text-error')
```
Comment (lha047) [inline src/ts/count-characters.ts:55]: Use the `color-error` class to get the red text

File: src/svelte/components/form/TextInput.svelte [hunk 7]
```diff
@@ -87,6 +90,9 @@
     id={name}
     {name}
     use:countCharacters={countCharsParams}
+    on:input={e => {
```
Comment (lha047) [inline src/svelte/components/form/TextInput.svelte:93]: Add a test for this new feature

File: src/svelte/components/form/TextArea.svelte [hunk 8]
```diff
@@ -53,6 +56,9 @@
   id={name}
   {name}
   use:countCharacters={countCharsParams}
+  on:input={e => {
```
Comment (lha047) [inline src/svelte/components/form/TextArea.svelte:59]: Add a test for this new feature

**Review by lha047 — CHANGES_REQUESTED** [review]
Left some comments. Sorry didnt see this the first time arround

**Comment by amundlindstol** [comment]
I personally think the code duplication is ok

**Review by lha047 — APPROVED** [review]
Nice good job :D

**Pull request merged**